packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -164,43 +164,14 @@ def __new__(mcs, name: str, bases: Tuple, namespace: Dict, **kwargs: Any) -> Typ
         if ABC in bases:
             # abstract class, return
             return new_cls
-        if issubclass(new_cls, BaseTxPayload):
-            mcs.transaction_type_to_payload_cls[new_cls.__name__] = new_cls  # type: ignore
-            return new_cls
         if not issubclass(new_cls, BaseTxPayload):
             raise ValueError(  # pragma: no cover
                 f"class {name} must inherit from {BaseTxPayload.__name__}"
             )
-        new_cls = cast(Type[BaseTxPayload], new_cls)
-
-        transaction_type = str(mcs._get_field(new_cls, "transaction_type"))
-        mcs._validate_transaction_type(transaction_type, new_cls)
-        # remember association from transaction type to payload class
-        mcs.transaction_type_to_payload_cls[transaction_type] = new_cls
 
+        mcs.transaction_type_to_payload_cls[new_cls.__name__] = new_cls
```

Comment:
What I tried to use is the following, but it results in errors as we have several modules where we define payloads with the same name as defined elsewhere. However I believe that this is what we want, otherwise we cannot guarantee we can load the payload from bytes correctly.
```
cls = mcs.transaction_type_to_payload_cls.get(new_cls.__name__)
if cls and cls is not new_cls:
    raise ValueError
```

Reply:
I actually was casting to dataclass here initially, as it removed the need for devs to do this
`freezing_payload_class = dataclass(new_cls, frozen=True)`, however the linters don't like it; they don't understand that the class attribute are in fact parameters of the initializer.

Reply:
You highlight an important issue. We have the same question with Events. It would totally hinder composability if the developer had to think about clashes outside their own skill. So I propose we include two pieces of info into the mapping: class name & skill id (without hash). That way, we can guarantee uniqueness even when two payloads have the same class name in different skills.

Reply:
Then we end up with the same issue though, we just move it to the transaction type. Now you can name payload classes identically, but not their transaction type. What I'm saying in [my previous comment](https://github.com/valory-xyz/open-autonomy/pull/1734#discussion_r1067084050) is that currently we can overwrite the payload class in the metaclass registry when they have matching name and payload_attribute. This is problematic since we assume that transaction_type is unique in the application (e.g. when we check if a particular paylaod is allowed during a specific round).